### PR TITLE
[SC] Micro optimisations for SC script tests

### PR DESCRIPTION
**Pull request by codingupastorm**
Contributes towards https://stratisplatformuk.visualstudio.com/StratisBitcoinFullNode/_workitems/edit/4212

### CODE REVIEW

File: src/Stratis.SmartContracts.Core/SmartContractScript.cs [hunk 1]
```diff
@@ -1,43 +1,62 @@
...
 
-        private static bool TestFirstByte(Script script, byte opcode)
+        /// <summary>
+        /// Tests whether a script starts with any of the given opcodes.
+        /// </summary>
+        /// <param name="script">The script to test.</param>
+        /// <param name="opcodes">A list of opcodes in byte form to test against.</param>
+        /// <returns>True if the script starts with one of the given opcodes.</returns>
+        [NoTrace]
+        private static bool TestFirstByte(Script script, byte[] opcodes)
```
Comment (rowandh) [inline src/Stratis.SmartContracts.Core/SmartContractScript.cs]: Don't like using an array of opcodes here - this should just be a simple test. Code is harder to understand and negligible performance benefit. Plus an array allocation every time you call it with a single opcode.

Comment (codingupastorm) [reply src/Stratis.SmartContracts.Core/SmartContractScript.cs]: Ah yes, it was worse when it was doing tobytes every time but in this case it makes sense

Comment (rowandh) [reply src/Stratis.SmartContracts.Core/SmartContractScript.cs]: We could also pass the script bytes in and avoid the duplicate call to `ToBytes`.

Comment (codingupastorm) [reply src/Stratis.SmartContracts.Core/SmartContractScript.cs]: The unsafe ToBytes just returns the bytes array inside of the Script so it's chill

**Pull request merged**